src/utils/events.ts

```diff
@@ -0,0 +1,37 @@
+import {ref} from "./api";
+import {IRemoveEvent, Events, IAddEvent, IFireEvent} from "../types/events"
+
+const events = ref<Events>({
+    onError: {},
+})
+
+export default function globalEvents() { 
```

Comment:
Despite those being composables they do not need to be called inside a component so it isn't certain you will have an `onMount` or `onUnmount` handler to hook into.

`AddEvent` was made to be duplication friendly, if you attach an event handler a hundred times with the same key, it will only register one event handler instead of 100. 

`RemoveEvent` was made for being able to handle use cases that might want a temporary event handler. For instance, a debug component might want to add an event handler to the error event and then remove it upon its unmounting.

`HasEvent` was made for more debugging purposes and for verification of an eventhandler with a specific key being attached already.

`FireEvent` is primarily for internal use but I intended on exposing it for test uses as well, so end users can have it for testing purposes.

Reply:
Exposed `AddEventHandler` and `RemoveEventHandler` publically, the others were retained for internal and testing purposes.